examples/core/dof/app.js

```diff
@@ -383,17 +393,22 @@ export const animationLoopOptions = {
       dofFramebuffer,
       quadVertexArray,
       dofProgram,
-      timerElement: new TimerElement(_animationLoop)
+      statsWidget
     };
   },
 
-  onRender: ({gl, tick, width, height, aspect, projMat, viewMat, instancedCubes, sceneFramebuffer, dofFramebuffer, quadVertexArray, dofProgram, timerElement}) => {
+  onRender: ({gl, tick, width, height, aspect, projMat, viewMat, instancedCubes, sceneFramebuffer, dofFramebuffer, quadVertexArray, dofProgram, statsWidget, _animationLoop}) => {
 
     if (!isDemoSupported) {
           return;
     }
 
-    timerElement.update();
+    if (widgetUpdateCount++ % 60 === 10) {
+      statsWidget.update();
```

Comment:
Suggestions: 

* Maybe manage the throttling inside the `widget.update()`, so that we could just call update here and relieve the app of the burden?

* This "resetting business" seems a little tedious for the app? Could there be a `reset()` method that just calls reset on all the stats? If some stat needs to be manually managed, maybe allow an option `{autoReset: false}` when adding that stat to the `Stats` object?

Reply:
This makes a lot of sense. The throttling count could be an argument to the constructor. And a reset loop could be associated with each stat, similar to the formatters.